tracer/src/Datadog.Trace/Tracer.cs

```diff
@@ -299,7 +297,14 @@ internal Scope StartActive(string operationName, ISpanContext parent = null, str
         /// <returns>The newly created span</returns>
         ISpan IDatadogOpenTracingTracer.StartSpan(string operationName, ISpanContext parent, string serviceName, DateTimeOffset? startTime, bool ignoreActiveScope)
         {
-            return StartSpan(operationName, parent, serviceName, startTime, ignoreActiveScope);
+            if (ignoreActiveScope && parent == null)
+            {
+                parent = SpanContext.None;
+            }
+
+            var span = StartSpan(operationName, parent, startTime);
+            span.ServiceName = serviceName;
```

Comment:
TODO: looks into this. It could be a behavior change if we set the value to `null` late in the process of creating a span. Before this change we were doing something like `ServiceName = serviceName ?? DefaultServiceName`.

Reply:
Resolved by only setting `span.ServiceName` if `serviceName` is not `null`.